src/index.js

```diff
@@ -1,43 +1,192 @@
+        document.getElementById(`pTask${i}`).style.display = 'none';
+        document.getElementById(`updt${i}`).style.display = 'flex';
+      });
+      a.style.backgroundColor = 'rgba(190, 180, 176, 0.61)';
+      document.getElementById(`updt${i}`).style.backgroundColor = 'transparent';
+    });
+
+    a.addEventListener('mouseout', () => {
+      document.getElementById(`delete${i}`).style.display = 'none';
+      document.getElementById(`ellipse${i}`).style.display = 'flex';
+      document.getElementById(`pTask${i}`).style.display = 'flex';
+      document.getElementById(`updt${i}`).style.display = 'none';
+      a.style.backgroundColor = 'transparent';
+    });
+  });
+}
+
+const getTask = () => {
+  if (localStorage.getItem('tasks') === null) {
+    storedTasks = [];
+  } else {
+    storedTasks = JSON.parse(localStorage.getItem('tasks'));
+  }
 
-const listDisplay = () => {
   let display = '';
-  toDoTask.forEach((task) => {
-    display += `<li class="task-item-container" id=${task.index}>
+  storedTasks.forEach((task, index) => {
+    const completed = task.completed ? 'line-through' : '';
+
+    display += `<li class="${completed}" id=${index}>
+    <div class = "task-item-container" id="content-1">
     <div class="task-item">
-      <input type="checkbox" id=${task.index} ${task.completed ? 'checked' : ''}>
-      <label type="text" id="${task.index}" for="${task.description}">${task.description}</label>
+    <input type="checkbox" id="${index} ${task.completed ? 'checked' : ''}" class="check">
+    <label type="text" class="Label" id="pTask${index}">${task.description}</label>
+    <input type="text" class="UpdateLabel" id="updt${index}"/>
+    </div>  
+    <div class="hide2" id="ellipse${index}">
+    <i class="fa-solid fa-ellipsis-vertical" ></i>
+    </div>
+    <div class="hide1" id="delete${index}">
+    <i id="idBtn${index}" class="fa-solid fa-trash-can" ></i>
     </div>
-    <i class="fa-solid fa-ellipsis-vertical"></i>
     </li>
     `;
   });
   toDoList.innerHTML = display;
+
+  showTrash();
+};
+
+const saveTask = ({ index, description, completed = false }) => {
+  storedTasks = [];
+
+  if (localStorage.getItem('tasks') === null) {
+    storedTasks = [];
+  } else {
+    storedTasks = JSON.parse(localStorage.getItem('tasks'));
+  }
+
+  storedTasks.push({ index, description, completed });
+  localStorage.setItem('tasks', JSON.stringify(storedTasks));
+  getTask();
+};
+
+const saveEditTask = (task) => {
+  const taskItem = storedTasks[task];
+  taskItem.description = addTaskInput.value;
+  localStorage.setItem('tasks', JSON.stringify(storedTasks));
+};
+
+addTask.addEventListener('click', (addItem) => {
+  addItem.preventDefault();
+  if (!addTaskInput.value) return;
+
+  if (editTaskItem != null) {
+    saveEditTask(editTaskItem);
+    editTaskItem = null;
+  } else {
+    saveTask({ index: storedTasks.length, description: addTaskInput.value, completed: false });
+  }
```

Comment:
- [ ] Kindly make sure the property index is set to the value of the new array length as described in the requirements(i.e. if a 5th task is added to the list, the index of that task should equal to the length of the new array (5)). 
- [ ] Currently a new task added gets an index of one less than the length of the new array.

Reply:
> * [ ]  Kindly make sure the property index is set to the value of the new array length as described in the requirements(i.e. if a 5th task is added to the list, the index of that task should equal to the length of the new array (5)).
> * [ ]  Currently a new task added gets an index of one less than the length of the new array.

As the previous reviewer suggested this change, Please fix it.

**Here is the project requirement:**
![r1](https://user-images.githubusercontent.com/70483835/166120688-6d220c61-b73d-4d4c-8499-f1fbcc89fe62.png)